### DISCUSSION TITLE
"credits" of each song in the API response

### DISCUSSION THREAD

**Discussion by areebvohra** [tidal-music/discussions#45] [Ideas & Feature requests]
It would be great if the API response would give us all the "credits" information like in the TIDAL App
- Producer
- Composer
- Lyricist
- A&R Director
- Assistant Engineer
- Associated Performer
- Background Vocal
- Engineer
- Executive Producer
- Mastering Engineer
- Programmer
- Recording Engineer
- Synthesizer
- Vocal Producer

![IMG_357F3981453B-1](https://github.com/tidal-music/discussions/assets/34376767/6e31027a-c580-46b1-a6f8-8687bf00b00f)

**Comment by stefanmoro** [comment]
Hey @areebvohra and thanks for the input!

Exposing credits together with the items in our catalogue is something we have discussed briefly internally, and that can open up for new use cases of our API. Glad to hear there is interest for this!
Although there is currently no timeline for how/when to expose this information, we will bring this request with us in our future discussions.

**Reply by areebvohra** [reply]
Hey @stefanmoro thank you for your response!

Is there any other way to get the credits like "Credits API" based on track_id or isrc.
The Track API has not been very helpful.

for example:
`https://openapi.tidal.com/tracks/byIsrc?isrc=USUM72120989&countryCode=US`
![Screenshot 2024-02-12 at 3 36 47 PM](https://github.com/tidal-music/discussions/assets/34376767/89343ff8-b1a7-4d55-829c-afb04de9bb69)

`https://openapi.tidal.com/tracks/261338003?countryCode=US`
![Screenshot 2024-02-12 at 3 35 51 PM](https://github.com/tidal-music/discussions/assets/34376767/2d89c5bb-1b07-451a-ba75-2d93c9e8112f)

**Comment by dansmith-kobalt** [comment]
We'd love to explore this too - interested to discuss what constraints there may be on Tidal side in surfacing this information from a rightsholder perspective (as a rightsholder ourselves).

**Comment by EljieGoc-ong** [comment]
Hi @stefanmoro is this now available on tidals v2 api?

**Comment by jettgalindo** [comment]
Hi @stefanmoro and TIDAL team! Adding a real-world use case to this thread since it's still the open request for credits in the API.
I'm a mastering engineer in Los Angeles. I built a small internal tool that verifies my own credits across platforms so my discography stays accurate, and TIDAL is one of the few services that displays full production credits in the app. That makes it one of the most valuable verification sources for those of us on the engineering side of records.

Right now that data isn't reachable for third-party apps. With a client-credentials app, /tracks/{id}?include=credits returns 200 but the credits relationship comes back empty, and the v1 credits endpoints return 401 (subStatus 6004). The OpenAPI spec marks the credits operations as INTERNAL access tier, so I understand this is intentional for now.

My use case isn't catalog harvesting. It's an engineer confirming her own name on her own work. Even a read-only, rate-limited path to credits would cover it. Is that something the team would consider, or is there an access tier I could apply for?

Happy to share more about the use case if helpful. Thanks for considering it!

**Comment by thesaycred** [comment]
@stefanmoro Commenting here since the past few responses have gone unanswered in this thread. Is this feature now available on Tidal's API?
